### DISCUSSION TITLE
How to disable MenuItem animation of HandyControls?

### DISCUSSION THREAD

**Discussion by GF-Huang** [HandyOrg/HandyControl#643] [Q&A] [answered]
@ghost1372 

![dfKeNgKZe3](https://user-images.githubusercontent.com/4510984/105632235-19222580-5e8d-11eb-8ca5-aae5df056797.gif)

![image](https://user-images.githubusercontent.com/4510984/105632273-32c36d00-5e8d-11eb-9423-be6194183553.png)

**Comment by ghost1372** [comment, marked as answer]
~~The only way right now is to override the style, Copy the following code into the app.xaml file~~

> I will add an AttachedProperty to disable the animation In the next version

Update:
I Added AttachedProperty
https://ghost1372.github.io/handycontrol/native_controls/menu/#Attached-Property

**Reply by GF-Huang** [reply]
Hey @ghost1372, I had try set `hc:MenuAttach.IsEnabledAnimation = False` global, but seems not work for the built-in `TextBox.ContextMenu`.

```xaml
<Style TargetType="Menu" BasedOn="{StaticResource MenuBaseStyle}">
  <Setter Property="hc:MenuAttach.IsEnabledAnimation" Value="False" />
</Style>

<Style TargetType="ContextMenu" BasedOn="{StaticResource ContextMenuBaseStyle}">
  <Setter Property="hc:MenuAttach.IsEnabledAnimation" Value="False" />
</Style>
``` 

![maybFBbrac](https://user-images.githubusercontent.com/4510984/113962044-912ba500-9859-11eb-8d7f-365bedf37918.gif)

**Reply by ghost1372** [reply]
You should use Textbox instead of Menu
```
<Style TargetType="TextBox" BasedOn="{StaticResource TextBoxBaseStyle}">
            <Setter Property="hc:MenuAttach.IsEnabledAnimation" Value="False" />
</Style>
```
this
`<Style TargetType="Menu" BasedOn="{StaticResource MenuBaseStyle}">`
only work for Direct use
![0](https://user-images.githubusercontent.com/9213496/114004944-5b47ea00-9874-11eb-83bf-cb6ae3a12012.gif)
